src/solver/sgd/forward_sgd.cpp

```diff
@@ -0,0 +1,165 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <miopen/conv_solution.hpp>
+#include <miopen/execution_context.hpp>
+#include <miopen/invoke_params.hpp>
+#include <miopen/tensor_view_utils.hpp>
+#include <miopen/sgd/solvers.hpp>
+
+#include <miopen/sgd/invoke_params.hpp>
+#include <miopen/datatype.hpp>
+#include <miopen/mlo_internal.hpp>
+#include <miopen/sgd.hpp>
+#include <miopen/target_properties.hpp>
+
+#define LOCAL_SIZE 256
+
+namespace miopen {
+namespace solver {
+namespace SGD {
+
+bool SGDForward::IsApplicable([[maybe_unused]] const ExecutionContext& constext,
+                              const miopen::SGD::ProblemDescription& problem) const
+{
+    if(!(problem.IsSameLength() && problem.IsSameType() && problem.IsValidLength()))
+    {
+        return false;
+    }
```

Comment:
Duplicated check with those in `ProblemDescription` constructor.

Reply:
"When the NetworkConfig is the same, IsApplicable() is not called. AMD recently requested that tensor consistency checks be done in ProblemDescription. " [source](https://moreh.atlassian.net/wiki/spaces/MV/pages/903512367/MV-GPU-MIOpen+NetworkConfig+example)

Reply:
removed check here